src/vs/workbench/contrib/chat/browser/chatWidget.ts

```diff
@@ -1050,17 +1047,15 @@ export class ChatWidget extends Disposable implements IChatWidget {
 				(timestamp, todayMs) => this.formatHistoryTimestamp(timestamp, todayMs),
 				todayMidnightMs
 			);
+			const listHeight = historyItems.length * 22;
+			welcomeHistoryList.style.height = `${listHeight}px`;
+			welcomeHistoryList.style.minHeight = `${listHeight}px`;
+			welcomeHistoryList.style.overflow = 'hidden';
 
-			// Set explicit height for the list to accommodate exactly the items we have
-			const listHeight = historyItems.length * 22; // 22px per item as defined in delegate
-			listEl.style.height = `${listHeight}px`;
-			listEl.style.minHeight = `${listHeight}px`;
-			listEl.style.overflow = 'hidden';
-
-			const historyList = this._register(this.instantiationService.createInstance(
+			this.historyList = this._register(this.instantiationService.createInstance(
```

Comment:
Isn't all this stuff leaking? Because the welcome view is rerendered many times, that's why we set it up with the `MutableDisposable`, but this is registering all these things to the ChatWidget and not disposing them. And I don't think we should recreate a whole list from scratch repeatedly to render this. There is much more opportunity to clean this up, if we're going to build it let's do it right.

Reply:
taking another look at this. i'll make sure we're not leaking first and since we have a template, i think we can just update parts when new sessions come in 👍🏻